src/main/java/org/sagebionetworks/web/server/servlet/AppConfigServlet.java

```diff
@@ -0,0 +1,155 @@
+
+  private static final Logger logger = Logger.getLogger(
+    AppConfigServlet.class.getName()
+  );
+
+  @Inject
+  public AppConfigServlet(
+    AWSAppConfigData appConfigDataClient,
+    StackConfiguration stackConfiguration
+  ) {
+    this.appConfigDataClient = appConfigDataClient;
+    this.stackConfiguration = stackConfiguration;
+  }
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    initializeAppConfigClient();
+  }
+
+  public void initializeAppConfigClient() {
+    if (appConfigDataClient == null) {
+      appConfigDataClient = AwsClientFactory.createAppConfigClient();
+    }
+    startConfigurationSession();
+    initializeConfigSupplier();
+  }
+
+  public void startConfigurationSession() {
+    try {
+      StartConfigurationSessionRequest sessionRequest =
+        new StartConfigurationSessionRequest()
+          .withApplicationIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-AppConfigApp"
+          )
+          .withEnvironmentIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-environment"
+          )
+          .withConfigurationProfileIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-configurations"
+          );
+      StartConfigurationSessionResult sessionResponse =
+        appConfigDataClient.startConfigurationSession(sessionRequest);
+      configurationToken = sessionResponse.getInitialConfigurationToken();
+    } catch (Exception e) {
+      configurationToken = null;
+    }
+  }
+
+  public void initializeConfigSupplier() {
+    configSupplier =
+      Suppliers.memoizeWithExpiration(
+        this::getLatestConfiguration,
+        5,
+        TimeUnit.MINUTES
+      );
+  }
+
+  public String getLatestConfiguration() {
+    if (configurationToken == null) {
+      return defaultConfigValue;
+    }
+    try {
+      GetLatestConfigurationRequest latestConfigRequest =
+        new GetLatestConfigurationRequest()
+          .withConfigurationToken(configurationToken);
+      GetLatestConfigurationResult latestConfigResponse =
+        appConfigDataClient.getLatestConfiguration(latestConfigRequest);
+      configurationToken = latestConfigResponse.getNextPollConfigurationToken();
+      ByteBuffer configData = latestConfigResponse.getConfiguration();
+      String newConfigValue = new String(
+        configData.array(),
+        java.nio.charset.StandardCharsets.UTF_8
+      ); // This may be empty if the client already has the latest version of the configuration.
+      if (!newConfigValue.isEmpty()) {
+        lastConfigValue = newConfigValue;
+      }
+      return lastConfigValue;
+    } catch (Exception e) {
+      return defaultConfigValue;
+    }
+  }
```

Comment:
I am worried about this case where the config is malformed JSON. I think the servlet should parse the JSON and handle the case where the JSON is malformed, and _never_ pass malformed JSON to the client.

WDYT of something like this, using `JSONObjectAdapter` to store data, therefore ensuring it is valid JSON?

```suggestion
  public JSONObjectAdapter getLatestConfiguration() {
    try {
      if (configurationToken == null) {
        return new JSONObjectAdapterImpl(defaultConfigValue);
      }
      GetLatestConfigurationRequest latestConfigRequest =
        new GetLatestConfigurationRequest()
          .withConfigurationToken(configurationToken);
      GetLatestConfigurationResult latestConfigResponse =
        appConfigDataClient.getLatestConfiguration(latestConfigRequest);
      configurationToken = latestConfigResponse.getNextPollConfigurationToken();
      ByteBuffer configData = latestConfigResponse
        .getConfiguration()
        .asReadOnlyBuffer();
      String newConfigString = new String(
        configData.array(),
        java.nio.charset.StandardCharsets.UTF_8
      );
      // This String may be empty if the client already has the latest version of the configuration.
      if (!newConfigString.isEmpty()) {
        lastConfigValue = new JSONObjectAdapterImpl(newConfigString);
      }
    } catch (Exception e) {
      logger.log(
        Level.SEVERE,
        "Failed to get or parse latest configuration",
        e
      );
    }
    return lastConfigValue;
  }
```